http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/db/NodepoolComponent.scala

```diff
@@ -86,27 +90,53 @@ object nodepoolQuery extends TableQuery(new NodepoolTable(_)) {
     nodepoolQuery
       .filter(_.id === id)
 
+  def saveAllForCluster(nodepools: List[Nodepool])(implicit ec: ExecutionContext): DBIO[Unit] =
+    for {
+      _ <- nodepoolQuery ++=
+        nodepools.map(fromNodepool(_))
+    } yield ()
+
   def saveForCluster(n: Nodepool)(implicit ec: ExecutionContext): DBIO[Nodepool] =
     for {
       nodepoolId <- nodepoolQuery returning nodepoolQuery.map(_.id) +=
-        NodepoolRecord(
-          NodepoolLeoId(0),
-          n.clusterId,
-          n.nodepoolName,
-          n.status,
-          n.auditInfo.creator,
-          n.auditInfo.createdDate,
-          dummyDate,
-          n.auditInfo.dateAccessed,
-          n.machineType,
-          n.numNodes,
-          n.autoscalingEnabled,
-          n.autoscalingConfig.map(_.autoscalingMin),
-          n.autoscalingConfig.map(_.autoscalingMax),
-          n.isDefault
-        )
+        fromNodepool(n)
     } yield n.copy(id = nodepoolId)
 
+  def markAsUnclaimed(ids: List[NodepoolLeoId])(implicit ec: ExecutionContext): DBIO[Unit] =
+    for {
+      _ <- nodepoolQuery
+        .filter(_.id.inSet(ids.toSet))
```

Comment:
Maybe use `inSetBind` -- IIRC `inSet` is the unsafe/unchecked version of the SQL `IN` clause